[IMP] website, *: consistent modules navigation

mano-odoo opened the pull request:
*: website_event, website_customer, website_crm_partner_assign, website_hr_recruitment

The goal is to provide visual consistency between sub navigation on website pages. 

- Avoiding different sized element between pages
- h2 title
- Improved semantic
- Ensuring same margins
- Responsive layout
- Offcanvas filter on mobile

task-4259418

Enterprise PR: https://github.com/odoo/enterprise/pull/74563


---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

anso-odoo reviewed (changes requested):
Hey @mano-odoo nice work.

I spotted some small stuff that could be improved.

Also, something more global: the the last `accordion-item` in the Events offcanvas has a border bottom while we remove it in other modules. IMO it works fine with the border, feel more structured and it would mean that we could remove some lines of SCSS. WDYT ?

| Events | Jobs |
|--------|--------|
| ![Screenshot 2024-12-20 at 16 55 30](https://github.com/user-attachments/assets/5a25eec9-3759-4fb5-9517-e0ccd77fbb90) | ![Screenshot 2024-12-20 at 17 07 05](https://github.com/user-attachments/assets/fa495c77-2f6c-4a50-9be8-ade8f4b69d2b) |

mano-odoo commented:
Hello @anso-odoo, it's actually an issue in /event since the last accordion item doesn't have a border in BS default. 
The border is coming from a placeholder element which shouldn't be there, making the styling rule fails. I modified the t-foreach and removed the placeholders elements to make it work as default.

anso-odoo reviewed (changes requested):
Hello @mano-odoo 

I figured that some `offcanvas` are wider than others on same resolution. Can you check to adapt ?

Example:
| Events | Customers |
|--------|--------|
| <img width="491" alt="Screenshot 2024-12-23 at 18 49 43" src="https://github.com/user-attachments/assets/863cbcd4-2de4-49e1-a1ff-ae96b524d672" /> | <img width="492" alt="Screenshot 2024-12-23 at 18 49 29" src="https://github.com/user-attachments/assets/7d2a33e9-b3e1-4c7b-8192-736247d9747a" /> |

anso-odoo reviewed (approved):
LGTM thanks!

SergeBayet commented:
@mano-odoo Could you rebase your branch before I start the review? Thanks :+1:

mano-odoo commented:
@SergeBayet there was a lot of changes on the /jobs page so it took a bit of time, it's ready for review ✅

SergeBayet commented:
> @SergeBayet there was a lot of changes on the /jobs page so it took a bit of time, it's ready for review ✅

Thanks @mano-odoo I'll look into it. It'll probably take a little while... :)

SergeBayet commented on addons/website_hr_recruitment/views/website_hr_recruitment_templates.xml:
```diff
@@ -773,91 +784,239 @@
+            </div>
+        </div>
+    </div>
 </template>
```
Can we do something to avoid this miswrap in mobile view?
![image](https://github.com/user-attachments/assets/0f369283-e4f3-4a70-b884-9ca53d91b9f5)
The events template is nicer:
![image](https://github.com/user-attachments/assets/5021a01f-f3fa-4fcc-91c1-33c188d4f507)

mano-odoo replied:
Woops forgot the `form-check` class, it should be good now

SergeBayet replied:
And (maybe overkill but why not), it seems that the number '4' is circled differently than the number '1'. Is that something easy to fix?

SergeBayet commented on addons/website_event/views/event_templates_list.xml:
```diff
@@ -46,102 +46,96 @@
+                                </ul>
                             </div>
-                        </t>
-                        <span class="o_wevent_offcanvas_date"/>
-                        <span class="o_wevent_offcanvas_country"/>
```
Are they not used by Bootstrap to handle the collapsing accordion? if you grep `o_wevent_offcanvas_date` for instance:
```xml
<xpath expr="//div[hasclass('accordion-item')]" position="after">
        <div class="accordion-item">
            <h2 class="accordion-header">
                <button class="accordion-button collapsed"
                    type="button"
                    data-bs-toggle="collapse"
                    data-bs-target=".o_wevent_offcanvas_date"
                    aria-expanded="false"
                    aria-controls="o_wevent_offcanvas_date">
                    Date
                </button>
                ...
```

mano-odoo replied:
It's confusing in that case because they had the same class but this one refers to the class on the `.accordion-collapse`
```xml
<div id="o_wevent_offcanvas_date" class="o_wevent_offcanvas_date accordion-collapse collapse" aria-labelledby="offcanvas_date">
```
Bootstrap uses it to know which collapse to open or close when we click on the button

SergeBayet replied:
Ok thanks! :)

mano-odoo commented:
@SergeBayet reminder 🙏

SergeBayet reviewed (approved):
LGTM :+1:  Thanks a lot for the work :)

sobo-odoo commented:
@robodoo delegate=SergeBayet

SergeBayet commented:
@robodoo rebase-ff

robodoo commented:
Merge method set to rebase and fast-forward.

robodoo commented:
@mano-odoo @SergeBayet linked pull request(s) odoo/enterprise#74563 not ready. Linked PRs are not staged until all of them are ready.

SergeBayet commented:
@robodoo r+
